bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.discovery;
+
+import java.util.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.boschshc.internal.devices.BoschSHCBindingConstants;
+import org.openhab.binding.boschshc.internal.devices.bridge.BridgeHandler;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Device;
+import org.openhab.binding.boschshc.internal.devices.bridge.dto.Room;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ThingDiscoveryService} is responsible discover Bosch Smart Home things.
+ * The SHC BridgeHandler is required to get the lists of rooms and devices.
+ * With this data the openhab things are discovered.
+ *
+ * @author Gerd Zanker - Initial contribution
+ */
+@NonNullByDefault
+public class ThingDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+    private static final int SEARCH_TIME = 1;
+
+    private final Logger logger = LoggerFactory.getLogger(ThingDiscoveryService.class);
+    private @Nullable BridgeHandler shcBridgeHandler;
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(
+            // BoschSHCBindingConstants.THING_TYPE_CLIMATE_CONTROL,
+            BoschSHCBindingConstants.THING_TYPE_INWALL_SWITCH,
+            // BoschSHCBindingConstants.THING_TYPE_MOTION_DETECTOR,
+            BoschSHCBindingConstants.THING_TYPE_SHUTTER_CONTROL,
+            // BoschSHCBindingConstants.THING_TYPE_THERMOSTAT,
+            BoschSHCBindingConstants.THING_TYPE_TWINGUARD
+    // BoschSHCBindingConstants.THING_TYPE_WINDOW_CONTACT
+    // TODO add other available SHC things
+    );
+
+    // @formatter:off
+    protected static final Map<String, String> DEVICEMODEL_TO_THINGTYPE_MAP = Map.ofEntries(
+            new AbstractMap.SimpleEntry<>("BBL", BoschSHCBindingConstants.THING_TYPE_SHUTTER_CONTROL.getId()), // Rollladenschalter
+            new AbstractMap.SimpleEntry<>("TWINGUARD", BoschSHCBindingConstants.THING_TYPE_TWINGUARD.getId()), // Twinguard
```

Comment:
Could we re-use existing constants instead of introducing a new one like "TWINGUARD"?

Reply:
I think no, because the existing strings like the ThinkTypeID in the BoschSHCBindingConstants are not matching to the required device model names from the SHC API.